packages/lesswrong/components/ea-forum/auth/EALoginPopover.tsx

```diff
@@ -0,0 +1,405 @@
+    marginBottom: 4,
+    fontSize: 14,
+    fontWeight: 500,
+  },
+  button: {
+    width: "100%",
+    height: 50,
+    padding: "15px 17px",
+    fontWeight: 600,
+    "& .Loading-spinner": {
+      height: "100%",
+      display: "flex",
+      alignItems: "center",
+    },
+  },
+  orContainer: {
+    display: "flex",
+    alignItems: "center",
+    gap: "11px",
+    width: "100%",
+    margin: "24px 0",
+    color: theme.palette.grey[600],
+  },
+  orHr: {
+    borderTop: `1px solid ${theme.palette.grey[600]}`,
+    flexGrow: 1,
+  },
+  socialContainer: {
+    display: "flex",
+    flexDirection: "column",
+    gap: "8px",
+    width: "100%",
+    marginBottom: 24,
+  },
+  socialButton: {
+    "& svg, & img": {
+      width: 20,
+      height: 20,
+      marginRight: 16,
+    },
+    "& .MuiButton-label": {
+      justifyContent: "flex-start",
+    },
+  },
+  switchPrompt: {
+    fontSize: 14,
+  },
+  switchPromptButton: {
+    color: theme.palette.primary.main,
+    fontWeight: 700,
+    userSelect: "none",
+  },
+  finePrint: {
+    color: theme.palette.grey[600],
+    marginTop: 16,
+    fontSize: 13,
+    fontWeight: 500,
+    lineHeight: "140%",
+    "& a": {
+      textDecoration: "underline",
+      "&:hover": {
+        textDecoration: "none",
+        opacity: 1,
+      },
+    },
+  },
+});
+
+const links = {
+  googleLogo: "/googleLogo.png",
+  eaOrg: "https://effectivealtruism.org",
+  terms: "/termsOfUse",
+  privacy: "https://ev.org/ops/about/privacy-policy",
+} as const;
+
+export const EALoginPopover = ({open, setAction, isSignup, classes}: {
+  open: boolean,
+  setAction: (action: "login" | "signup" | null) => void,
+  isSignup: boolean,
+  classes: ClassesType<typeof styles>,
+}) => {
+  const client = useAuth0Client();
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [showPassword, setShowPassword] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [resetPasswordLoading, setResetPasswordLoading] = useState(false);
+  const [message, setMessage] = useState<string | null>(null);
+  const [error, setError] = useState<string | null>(null);
+
+  const onChangeEmail = useCallback((ev: ChangeEvent<HTMLInputElement>) => {
+    setEmail(ev.target.value);
+  }, []);
+
+  const onChangePassword = useCallback((ev: ChangeEvent<HTMLInputElement>) => {
+    setPassword(ev.target.value);
+  }, []);
+
+  const toggleShowPassword = useCallback(() => {
+    setShowPassword((showPassword) => !showPassword);
+  }, []);
+
+  const onSubmit = useCallback(async (ev: FormEvent<HTMLFormElement>) => {
+    ev.preventDefault();
+
+    if (!email || !password) {
+      setError("Email and password are required");
+      return;
+    }
+
+    setMessage(null);
+    setError(null);
+
+    try {
+      setLoading(true);
+      await (
+        isSignup
+          ? client.signup(email, password)
+          : client.login(email, password)
```

Comment:
nit: If you try to sign up with a password that is too weak it just says "password two weak", rather than saying what needs to be added